checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/rule/ActiveRuleWrapperBatchImpl.java

```diff
@@ -0,0 +1,58 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.sonar.plugins.checkstyle.rule;
+
+import java.util.Map;
+
+import org.sonar.api.batch.rule.ActiveRule;
+import org.sonar.plugins.checkstyle.CheckstyleSeverityUtils;
+
+public class ActiveRuleWrapperBatchImpl implements ActiveRuleWrapper {
```

Comment:
we need to add javadoc and explain that we need such classes only till we upgrade to 7.X LST.
and some details as you did in PR comment.

Reply:
I added a JavaDoc comment for the interface and here as a short reference to what the wrapper is wrapping.
However, as noted in the other JavaDoc, for now we have to use the wrapper unless we refactor writing the checkstyle configuration from the exporter which needs `RulesProfile` and we don't want to copy paste the entire class only for the new `ActiveRules` scanner side part.

Reply:
please  create issue on this.